libvirt/resource_libvirt_network_test.go

```diff
@@ -48,6 +48,48 @@ func TestNetworkAutostart(t *testing.T) {
 	})
 }
 
+func TestNetworkDNS(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLibvirtNetworkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+				resource "libvirt_network" "test_net" {
+					name      = "networktest"
+					mode      = "nat"
+					domain    = "k8s.local"
+					addresses = ["10.17.3.0/24"]
+					dns {
```

Comment:
for this kind of testing we need to parse the networking xml definitio of the network and check if the xml entity is set correctly.

Then we can assume that it works correctly, i will post you an example soon, i"m dooing it for the dhcp.

Reply:
any luck on a pointer to a real test?

Reply:
@eparis  sorry for delay yop, look here: 

https://github.com/dmacvicar/terraform-provider-libvirt/pull/385/files#diff-620e6bdaa84869e53998f4dbbdf65204R63

I will help you more on this pr next week :rose: :sunrise:

Reply:
That's awesome, I'll check it out!